### PR TITLE
Implement blockchain reactor for fast-syncing purpose

### PR DESCRIPTION
**Pull request by y4t0g4m1**
Implementation details of these 4 modules below is described at https://github.com/kardiachain/go-kardia/issues/131.
- [X] Blockchain reactor
- [X] Routine
- [X] Processor
- [X] Scheduler
- [X] Add fast sync config to node configs
```
FastSync:
  ServiceName: BCREACTOR # log tag of blockchain reactor logs. Type string
  Enable: true                            # true if this node allow and be able to fastsync, otherwise false
  MaxPeers: 2                           # maximum peer is allowed to receive fastsync blocks from this node at a time. Type int
  TargetPending: 20                  # maximum number of blocks in a batch sync. Type int
  PeerTimeout: 15                     # maximum response time from a peer in second. Type int
  MinRecvRate: 0                      # minimum receive rate from peer, otherwise prune. Type int64
```

### CODE REVIEW

**Review by lewtran — CHANGES_REQUESTED** [review]
Add implementation descriptions for each sections

File: behaviour/doc.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+4. block part
+
+type blockPart struct {
+	explanation string
+}
+
+This message will request the peer be marked as good
+
+*/
+package behaviour
```
Comment (lewtran) [inline behaviour/doc.go:42]: - Move 'behaviour' package to lib/p2p/behaviour
- Move this to Doc.md

**Review by lewtran — CHANGES_REQUESTED** [review]

File: types/block.go [hunk 2]
```diff
@@ -309,6 +309,7 @@ func NewBlock(header *Header, txs []*Transaction, lastCommit *Commit, evidence [
 		b.header.EvidenceHash = common.NewZeroHash()
 	}
 
+	b.Hash()
```
Comment (lewtran) [inline types/block.go:312]: Add comment and define its purpose

**Review by lewtran — CHANGES_REQUESTED** [review]

File: types/block.go (hunk 2 above)
Comment (y4t0g4m1) [reply types/block.go:312]: I removed it. Fixed in blockchain reactor test suite.

File: behaviour/doc.go (hunk 1 above)
Comment (y4t0g4m1) [reply behaviour/doc.go]: Moved it!

**Review by lewtran — APPROVED** [review]

**Pull request merged**